core/src/main/java/org/ldaptive/pool/AbstractConnectionPool.java

```diff
@@ -1061,56 +1063,30 @@ public void prune()
     poolLock.lock();
     try {
       if (!available.isEmpty()) {
-        final int currentPoolSize = active.size() + available.size();
-        if (currentPoolSize > minPoolSize) {
-          logger.debug("Pruning available pool of size {} for {}", available.size(), this);
-
-          final int numConnAboveMin = currentPoolSize - minPoolSize;
-          final int numConnToPrune = available.size() < numConnAboveMin ? available.size() : numConnAboveMin;
-          final List<Callable<PooledConnectionProxy>> callables = new ArrayList<>(numConnToPrune);
-          for (PooledConnectionProxy pc : available) {
-            logger.trace("pruning {} for {}", pc, this);
-            callables.add(() -> {
-              if (pruneStrategy.apply(pc)) {
-                logger.trace("prune approved on {} with {} for {}", pc, pruneStrategy, AbstractConnectionPool.this);
-                return pc;
-              }
+        int numConnPruned = 0;
+        final int size = pruneStrategy.getPrunePredicates().size();
+        logger.trace("prune strategy has {} predicates", size);
+        // prune all available connections for each predicate supplied by the prune strategy
+        for (int i = 0; i < size; i++) {
```

Comment:
This outer loop seems unnecessary. As I noted elsewhere I don't understand why you need to supply the index into the predicates list if they're evaluated in order. Seems like you could just iterate over them with an iterator as you do in the inner loop here.

Reply:
Each predicate must be invoked for each connection in the pool, in order.
The outer loop iterates over the predicates, the inner loop iterates over the pool.

Reply:
I see, thanks